controllers/artefacts.rb

```diff
@@ -57,6 +57,111 @@ class ArtefactsController < ApplicationController
 
+        # Ressources
+        namespace "/:artefactID/resources" do
+            get do
+                ontology, submission = get_latest_submission
+                _, page, size = settings_params(LinkedData::Models::Class).first(3)
+                size_per_route = (size / 5).to_i
+                
+                resource_types = [
+                  LinkedData::Models::Class,
+                  LinkedData::Models::Instance,
+                  LinkedData::Models::SKOS::Scheme,
+                  LinkedData::Models::SKOS::Collection,
+                  LinkedData::Models::SKOS::Label
+                ]
+
+                resources = resource_types.flat_map do |model|
+                  handle_resource_request(model, page, size_per_route, model.goo_attrs_to_load([])).to_a
+                end
+
+                reply Goo::Base::Page.new(page, size, resources.length, resources)
+            end
+          
+            get '/classes' do
+                ontology, submission = get_latest_submission
+                type = LinkedData::Models::Class.class_rdf_type(submission)
+                attributes, page, size = settings_params(LinkedData::Models::Class).first(3)
+                
+                if type == RDF::OWL[:Class]
+                    attributes.reject! { |x| x.is_a?(Hash) }
+                    reply handle_resource_request(LinkedData::Models::Class, page, size, attributes)
+                else
+                    reply empty_page(page, size)
+                end
+            end
+          
+            get '/concepts' do
+                ontology, submission = get_latest_submission
+                type = LinkedData::Models::Class.class_rdf_type(submission)
+                attributes, page, size = settings_params(LinkedData::Models::Class).first(3)
+                
+                if type.to_s == "http://www.w3.org/2004/02/skos/core#Concept"
+                    attributes.reject! { |x| x.is_a?(Hash) }
+                    reply handle_resource_request(LinkedData::Models::Class, page, size, attributes)
+                else
+                    reply empty_page(page, size)
+                end
+            end
+          
+            get '/properties' do
+                ont, latest_submission = get_latest_submission
+                begin
+                    props = ont.properties(latest_submission)
+                rescue LinkedData::Models::Ontology::ParsedSubmissionError => e
+                    error 404, e.message
+                end
+                
+                _, page, size = settings_params(LinkedData::Models::OntologyProperty).first(3)
+                reply Goo::Base::Page.new(page, size, props.length, props)
+            end
+          
+            %w[individuals schemes collections labels].each do |resource_type|
+                get "/#{resource_type}" do
+                    model_class = case resource_type
+                        when 'individuals' then LinkedData::Models::Instance
```

Comment:
Scheme, collections, labels, need to be empty in OWL
Individuals need to be empty in SKOS